### PR TITLE
Clean up and make more image handling consistent

### PR DESCRIPTION
**Pull request by johnSchnake**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
- Cleans up some of the image registry handling by
initializing values in a more consistent and clear
manner.

- Adds the Docker library registry to the list of
values that is override-able.

- Adds a few branches to logic to ensure each registry
is handled the same.

- Adds testing to cover new cases

#### Which issue(s) this PR fixes:

Partially addresses #100489

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: build/dependencies.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ dependencies:
     version: "2.31"
     refPaths:
     - path: test/utils/image/manifest.go
-      match: configs\[Agnhost\] = Config{promoterE2eRegistry, "agnhost", "\d+\.\d+"}
+      match: configs\[Agnhost\] = Config{list\.PromoterE2eRegistry, "agnhost", "\d+\.\d+"}
```
Comment (johnSchnake) [inline build/dependencies.yaml:33]: This file just had to be updated to match the new line of code; no functional change.

File: test/utils/image/manifest.go [hunk 2]
```diff
@@ -42,6 +42,7 @@ type RegistryList struct {
 	PrivateRegistry         string `yaml:"privateRegistry"`
 	SampleRegistry          string `yaml:"sampleRegistry"`
 	MicrosoftRegistry       string `yaml:"microsoftRegistry"`
+	DockerLibraryRegistry   string `yaml:"dockerLibraryRegistry"`
```
Comment (johnSchnake) [inline test/utils/image/manifest.go:45]: As per the issue, you may need to override this value so I don't see why it isnt customizable.

File: test/utils/image/manifest.go [hunk 3]
```diff
@@ -99,26 +88,26 @@ func initReg() RegistryList {
...
-	gcAuthenticatedRegistry = registry.GcAuthenticatedRegistry
-	gcEtcdRegistry          = registry.GcEtcdRegistry
-	gcRegistry              = registry.GcRegistry
-	sigStorageRegistry      = registry.SigStorageRegistry
-	gcrReleaseRegistry      = registry.GcrReleaseRegistry
-	invalidRegistry         = registry.InvalidRegistry
-	sampleRegistry          = registry.SampleRegistry
-	microsoftRegistry       = registry.MicrosoftRegistry
-
-	imageConfigs, originalImageConfigs = initImageConfigs()
```
Comment (johnSchnake) [inline test/utils/image/manifest.go:121]: This is a section that exemplifies the cleanup:
 - most values were hardcoded above this, so just refer to that value
 - docker wasn't defined above so thats hardcoded here
 - most are unexported but PrivateRegistry is, even though its not used elsewhere.

File: test/utils/image/manifest.go [hunk 4]
```diff
@@ -358,8 +347,15 @@ func GetPauseImageName() string {
 	return GetE2EImage(Pause)
 }
 
-// ReplaceRegistryInImageURL replaces the registry in the image URL with a custom one
+// ReplaceRegistryInImageURL replaces the registry in the image URL with a custom one based
+// on the configured registries.
 func ReplaceRegistryInImageURL(imageURL string) (string, error) {
+	return replaceRegistryInImageURLWithList(imageURL, registry)
```
Comment (johnSchnake) [inline test/utils/image/manifest.go:353]: Exported method uses the package level variable; the logic was extracted into an unexported method where we can pass the RegistryList for easier testing/injection.

File: test/utils/image/manifest_test.go [hunk 5]
```diff
@@ -24,108 +24,86 @@ import (
...
-	},
-	{
-		"unknwon.io/google-samples/test:latest",
-		result{
-			result: "",
-			err:    fmt.Errorf("Registry: unknwon.io/google-samples is missing in test/utils/image/manifest.go, please add the registry, otherwise the test will fail on air-gapped clusters"),
-		},
-	},
-}
-
```
Comment (johnSchnake) [inline test/utils/image/manifest_test.go:107]: Just shortened all this and put it in the test itself. The way it was written seems like those structures/values needed to be reused which isn't true and made this section harder to read.

**Comment by johnSchnake** [comment]
Self-commented to help (hopefully) speed up review.

**Comment by hh** [comment]
+1
/lgtm

**Review by justaugustus — APPROVED** [review]
Thanks @johnSchnake!
/lgtm
/approve
/retest

**Pull request merged**